lib/presentation/features/home/bloc/home_bloc.dart

```diff
@@ -1,28 +1,47 @@
+import 'package:flutter/widgets.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:marvelapp_flutter/domain/use_cases/get_characters_use_case.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_event.dart';
 import 'package:marvelapp_flutter/presentation/features/home/bloc/home_state.dart';
 import 'package:marvelapp_flutter/presentation/extensions/to_character_view_data.dart';
+import 'package:marvelapp_flutter/presentation/models/character_view_data.dart';
 
 class HomeBloc extends Bloc<HomeEvent, HomeState> {
   final GetCharactersUseCase getCharactersUseCase;
 
-  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: true)) {
+  HomeBloc({required this.getCharactersUseCase}) : super(const HomeState(loading: false)) {
     on<GetHeroes>(_mapGetHeroesEventToState);
   }
 
   Future<void> _mapGetHeroesEventToState(GetHeroes event, Emitter<HomeState> emit) async {
-    emit(state.copyWith(loading: true, error: null));
+    if (state.hasReachedMax) return;
+    if (state.loading) return;
```

Comment:
This loading means the loading of the first page, right? When you loading the following pages is it false? If so - it is confusing.

Reply:
done